.github/workflows/pylint.yml

```
@@ -0,0 +1,45 @@
+
+
+name: Pylint
+
+on:
+  workflow_dispatch:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8  # maybe better not hardcoded
+      - name: Install and configure poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+      - name: Disable modern installer
+        run: poetry config installer.modern-installation false
+      - name: Load cached venv
+        id: cached-pip-wheels
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Install dependencies
+        run: poetry install --no-interaction --no-root
+        if: steps.cache.outputs.cache-hit != 'true'
+      - name: Install project
+        run: poetry install --no-interaction
+      - name: Install task runner
+        run: pip install poethepoet
+      - name: Lint
+        run: |
+          source $VENV
+          poe lint
+          poe lint-warnings
```

poethepoet (`poe`) is already making use of the correct environment, so you do not need activation.

Moreover, the `$VENV` variable is most likely unset, so you are sourcing nothing.

```suggestion
          poe lint
          poe lint-warnings
```